prediction_market_agent_tooling/markets/polymarket/copy_trading.py

```
@@ -0,0 +1,461 @@
+        self._state.last_poll_timestamp = utcnow()
+        if self.state_file_path:
+            self.save_state()
+
+        return results
+
+    def run_loop(self, run_time: float | None = None) -> None:
+        start_time = time.time()
+        while run_time is None or time.time() - start_time < run_time:
+            try:
+                self.run_once()
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)